common/network-common/src/main/java/org/apache/spark/network/shuffledb/LevelDB.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.shuffledb;
+
+import org.iq80.leveldb.DBIterator;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LevelDB implements DB {
```

Comment:
I'm wondering if we can reuse the following.

https://github.com/apache/spark/blob/bf8a4c47ac3752edf86f1e14e4050e8d202b34a4/common/kvstore/src/main/java/org/apache/spark/util/kvstore/LevelDB.java#L46

Reply:
`KVStore` currently provides a slightly higher level abstraction - while existing implementation in shuffle does a more low level use of LevelDB :-)

Reply:
+1 to the idea of leveraging existing code in kvstore. Right now, this part of the code is only being used by HistoryServer. But I think it is reasonable to either extend or refactor to make it being used for DBs in Shuffle.

Reply:
a little difficult, as @mridulm said:

> KVStore currently provides a slightly higher level abstraction - while existing implementation in shuffle does a more low level use of LevelDB :-)